tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (someBool: boolean) => {
+        if(someBool) {
+          return Promise.reject("error");
+        }
+      }
+    };
+
+    const validAnyWithBoolRestriction = validModel(anyWithBoolRestriction);
+
+    it('should fail against the restriction', function(done) {
+      mochaAssertPromiseErrorsWith(
+        validAnyWithBoolRestriction(true),
+        (error) => {
+          return error === "error"
+        },
         done
-      );
+      )
+    });
+
+    const anyWithLengthRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (arrayOfNumber: number[]) => {
```

Comment:
This could be more than an array, `{ length: 5 }` should pass the validation as well. Instead of `arrayOfNumber: number[]`, go with `{ anyWithLength: { length: number }}`